.github/workflows/common/common_python_tox_lint.yml

```diff
@@ -0,0 +1,37 @@
+# A common CI workflow to run linting as defined in tox.ini
+#
+# Assumptions:
+#
+# 1. The repo has a tox.ini file, which defines a 'lint' command.
```

Comment:
Do we really want to be promoting Tox? It's a rather painful tool in my experience. Most of the time when I run it for the first time in a new repository it fails in some obscure way because my Python installation doesn't match that of the one who set up Tox for the repository.

That's not really Tox's fault, it's more of an indictment against Python packaging and version management as a whole, but it's still a poor experience for contributors.

Reply:
@WillDaSilva Is there an alternative you prefer?

Reply:
For linting? I've been finding `pre-commit` pretty good for that job, but we don't need a workflow for that since we could use https://pre-commit.ci instead.

In either case, you need to ensure you're running on the right Python version, whatever that may be for a given project. Here we assume that's Python 3.10.

Reply:
Yeah I'm 100% in favor of replacing Tox with pre-commit for all lint checks & fixes

Reply:
No, this is just legacy. The "--capture=no" translates basically to "print=all" as you say above, and I agree that's probably not what we want here. (For security reasons as well as log length reasons.)

Reply:
I think there's a way to manually invoke the equivalent of what the cloud runs with pre-commit.ci, yes? 

I'm a bit behind on that invocation, and might need to do a deeper dive / refresher.

Reply:
For now, I think I'm in favor of committing this lint workflow since it pairs with our cookiecutter. I've already named the file in a way that it is specific to this invocation pattern. We can just not enable it when it isn't useful, but in the meanwhile, we still have the option to do so for repos where the tox test exists and the pre-commit.ci set does not (yet).

Reply:
> I think there's a way to manually invoke the equivalent of what the cloud runs with pre-commit.ci, yes?

`pre-commit run --all-files`

n.b. the default `pre-commit` / `pre-commit run` command will only run on the staged changes.

Reply:
> We can just not enable it for repos where it isn't useful, but in the meanwhile, we still have the option to do so for repos where the tox-based lint rules exist and the pre-commit.ci set does not (yet).
> 
> What do you think of that, at least for short-term?

Make sense :+1: